packages/plugins/gateway/src/stitching.ts

```diff
@@ -0,0 +1,95 @@
+import { print, GraphQLSchema } from 'graphql';
+import { SchemaChangeCallback, GatewayOrchestrator } from './types';
+import { stitchSchemas } from '@graphql-tools/stitch';
+import { introspectSchema, makeRemoteExecutableSchema } from '@graphql-tools/wrap';
+import { fetch } from 'cross-fetch';
+import { AsyncExecutor } from '@graphql-tools/utils';
+
+export type SchemaLoader = {
+  loadSchema(): Promise<GraphQLSchema>;
+};
+
+export type NotifyChangedFn = () => void;
+export type ReloadProbe = (reload: NotifyChangedFn) => void;
+
+export type Service = {
+  name: string;
+  loader: SchemaLoader;
+  reloadProbe?: ReloadProbe;
+};
+
+export function pollingProbe(options: { interval: number }): ReloadProbe {
+  return runFn => {
+    setTimeout(() => {
+      console.log('INTERVAL');
+      runFn();
+    }, options.interval);
+  };
+}
+
+export function endpoint(config: { url: string }): SchemaLoader {
+  const executor: AsyncExecutor = async function remoteExecutor({ document, variables }) {
+    const query = print(document);
+    const fetchResult = await fetch(config.url, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ query, variables }),
+    });
+
+    return fetchResult.json();
+  };
+
+  return {
+    loadSchema: async () => {
+      const schema = await introspectSchema(executor);
+
+      return makeRemoteExecutableSchema({
```

Comment:
This method is deprecated. We return the subschemaconfig directly to stitchSchemas to prevent multiple wrapping and reduce delegation rounds